networking/interconnect/main.tf

```diff
@@ -0,0 +1,99 @@
+
+/**
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+resource "google_compute_router" "router" {
+ 
+    name    = var.router_name == "" ? "router-${var.vlan_attachment_name}" : var.router_name
+    description=var.router_description
+    project = var.project_id
+    region = var.region
+    network = var.network_name
+  
```

Comment:
Please don't leave extra newlines and run files through `terraform fmt` before submitting.